---
title: "Cleaning"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Read in data
```{r}
#rm(list = ls())
library(tidyverse)
beta <- readRDS("/nfs/turbo/bakulski1/People/alreiner/data/betaqc.rds")
#str(beta)
#dim(beta)
#colnames(beta)

fullphen <- readRDS("/nfs/turbo/bakulski1/People/alreiner/data/fullpheno.rds")
#str(fullphen)

pdqc <- readRDS("/nfs/turbo/bakulski1/People/alreiner/data/pdqc.rds")
```

```{r}
# create why missing variable
# get methylation data into 1 dataset first

#Factor key categorical variables
# fullphen$m1city <- factor(fullphen$m1city)
# fullphen$cm1bsex <- factor(fullphen$cm1bsex)
# fullphen$cm1edu <- factor(fullphen$cm1edu)
# fullphen$cm1ethrace <- factor(fullphen$cm1ethrace)
# fullphen$m1g1 <- factor(fullphen$m1g1)
# fullphen$m1g4 <- factor(fullphen$m1g4)
# pdqc$cm1ethrace <- factor(pdqc$cm1ethrace)

# get rid of the moms and teens in pdqc
pdqc1 <- pdqc %>%
  dplyr::filter(childteen != "T") %>%
  filter(childteen != "M")
#n=866 (1 case is not in wave)

# returns logical T/F if any duplicate C and idnum combinations (paste0--combines C/T and idnum)
pdqcA <- pdqc1 %>%
  #group_by(idnum)%>%
  #mutate(min = min(probe_fail_pct))%>%
  #arrange(probe_fail_pct)%>%
  mutate(crossdupe = duplicated(paste0(pdqc1$childteen,pdqc1$idnum)),
         flag_sex = predicted_sex != sex
         )%>%
  select(1:19, -cm5intyr, -cm5intmon, -cmf5finjail,crossdupe, probe_fail_pct,flag_sex)
   #filter(idnum %in% idnumsdup)
sum(pdqcA$flag_sex==T)

#create flag variable for people with discordant sex
idnums_sex_dis <- pdqcA %>% filter(flag_sex==T)%>%select(idnum)
idnums_sex <- c(idnums_sex_dis$idnum)
head(pdqc$predicted_sex)
head(pdqc$sex)
head(pdqc$cm1bsex)

pdqcA1 <- pdqcA %>%
group_by(idnum)%>%
  mutate(min = min(probe_fail_pct))%>%
  arrange(probe_fail_pct)%>%
  mutate(crossdupety = duplicated(paste0(childteen,idnum)))
#%>%filter(idnum %in% idnumsdup) to check that the sample where crossupety=T (which will be removed is the higher probe_pct_fail value)

pqr <- pdqcA1 %>%
  filter(crossdupety==T)

# save idnums of these technical duplicates
idnumsdup <- c(pqr$idnum)

#check how many duplicate C,idnum (n=25)
sum(pdqcA1$crossdupety==T)

#view probe fail pct for these idnums
lmno <- pdqcA %>%
  filter(idnum %in% idnumsdup)%>%
  select(idnum, probe_fail_pct, crossdupe,childteen)

# scatterplot of probe_pct fails for each duplicate--see how similar the quality of reads is
it1 = lmno%>%filter(crossdupe==F)
it2 = lmno%>%filter(crossdupe==T)

plot(it1$probe_fail_pct, it2$probe_fail_pct, ylim=range(it1$probe_fail_pct), xlab="Probe Fail Percent Iteration 1",ylab="Probe Fail Percent Iteration 2", main="Sample quality comparison for n=25 technical duplicates")+abline(0,1)
#sort(pdqcA$probe_fail_pct)

#keep the lower value of the 2
g<- lmno %>%
  group_by(idnum)%>%
  mutate(min = min(probe_fail_pct))%>%
  arrange(probe_fail_pct)%>%
  mutate(crossdupet = duplicated(paste0(childteen,idnum)))
  
# # scatterplot of probe_pct fails for each duplicate NOW making x-axis the minimum probepctfails and y-axis the samp with larger probe_fail_pct 
itsm = g%>%filter(crossdupet==F)
itlg = g%>%filter(crossdupet==T)
  
plot(itsm$probe_fail_pct, itlg$probe_fail_pct, xlim=range(itlg$probe_fail_pct),xlab="Minimum Probe Fail Percent",ylab="Greater Probe Fail Percent", main="Sample quality comparison for n=25 technical duplicates")+abline(0,1)
#xlim=c(0,0.04), ylim=c(0,0.4)

#remove the duplicates
pdqcAA <- pdqcA[!pdqcA$crossdupe,] #only keeps rows (samples) where FALSE (non- duplicates) # 841
# CORRECTION: you still have one of the samples from the 25 duplicates!!



#gaphunter

#820
# gaps820 <- gaphunter(beta[,match(mvFIN_clean$MethID,colnames(beta))], threshold=0.05, keepOutliers = FALSE, outCutoff=0.01, verbose=TRUE)
# 
# gapprobes820 <- c(rownames(gaps820$proberesults))
# head(gaps820$proberesults)

```


```{r}

# Load data listing location of each Cpg site (shore, shelf, etc)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
table(Islands.UCSC$Relation_to_Island)
data("Islands.UCSC")

# Try to see regions of CpG sites in our beta
Islands.UCSC_beta <- Islands.UCSC[rownames(beta),]
sum(duplicated(rownames(Islands.UCSC_beta)))

# now use 
Islands.UCSC_beta_XY <-  as.data.frame(Islands.UCSC_beta) %>% dplyr::filter(grepl("[XY]",Islands.UCSC_beta$Islands_Name))

table(unique(Islands.UCSC_beta_XY$Islands_Name))

CpGxy <- c(rownames(Islands.UCSC_beta_XY))

#try new filter: x chrom problem
Islands.UCSC_betanogap <- Islands.UCSC[rownames(beta_nogap),]

#try remove open sea sites
Islands.UCSC_betanogapnoXY <- Islands.UCSC[rownames(beta_nogap_noXY),]
table(Islands.UCSC_betanogapnoXY$Relation_to_Island) #nearly 34% open sea that we don't know whether they are autosomal or gap probes

Islands.UCSC_betanogapnoXY_OS <- as.data.frame(Islands.UCSC_betanogapnoXY)%>% dplyr::filter(Relation_to_Island == "OpenSea")
OSsites <- c(rownames(Islands.UCSC_betanogapnoXY_OS))




#can check if all rownmaes of betanogap are in the dataset?
print("ch.X.7566543R" %in% rownames(Islands.UCSC_betanogap))
print("ch.X.147603R" %in% rownames(Islands.UCSC_betanogap))
print("ch.X.223217F" %in% rownames(Islands.UCSC_betanogap))

# what are their island name labels in islands
View(as.data.frame(Islands.UCSC_betanogap[c("ch.X.7566543R","ch.X.147603R","ch.X.223217F","ch.X.153018199F"),]))

# how many are missing island name
View(as.data.frame(is.na(Islands.UCSC_betanogap$Islands_Name)))
View(as.data.frame(Islands.UCSC_betanogap))
length(which(Islands.UCSC_betanogap$Islands_Name == " "))

View(Islands.UCSC_betanogap[which(Islands.UCSC_betanogap$Islands_Name == " "),])

View(as.data.frame(Islands.UCSC_beta[which(Islands.UCSC_beta$Islands_Name == " "),]))
View(as.data.frame(Islands.UCSC[which(Islands.UCSC$Islands_Name == " "),]))

which(Islands.UCSC$Islands_Name =="")
# it says FALSE when using is.na() on island names
# but there are many Cpg Sites missing island name information omg

# see if the entries missing islands name regions in our betanogap have info in og Islands.UCSC data
View(as.data.frame(Islands.UCSC[c("cg02393514","cg01850541","cg01573544"),]))
sum(is.na(Islands.UCSC$Islands_Name))

length(which(Islands.UCSC$Islands_Name =="")) #176,047 sites do not have island name entries!

View(as.data.frame(Islands.UCSC[which(Islands.UCSC$Islands_Name ==""),]))



Islands.UCSC_beta_XY_check <-  as.data.frame(Islands.UCSC_betanogap) %>% dplyr::filter(grepl("[XY]",Islands.UCSC_betanogap$Islands_Name))
View(Islands.UCSC_beta_XY_check) # no missing island values here b/c filtered them out!
#or has x in the column name

#as.data.frame(table(unique(rownames(Islands.UCSC_beta_XY_check))))

#Wrong sites
# CpGxy <- c(rownames(Islands.UCSC_beta_XY))
# 
# 
# # just x chrom sites in Island
# Islands.UCSC_beta_X <-  as.data.frame(Islands.UCSC_beta) %>% dplyr::filter(grepl("X",Islands.UCSC_beta$Islands_Name))
# 
# CpGx <- c(rownames(Islands.UCSC_beta_X))
# 
# # just y chrom sites in Island
# Islands.UCSC_beta_Y <-  as.data.frame(Islands.UCSC_beta) %>% dplyr::filter(grepl("Y",Islands.UCSC_beta$Islands_Name))
# 
# CpGy <- c(rownames(Islands.UCSC_beta_Y))
# 
# ## 7051  sites map to X chromosome
# head(sum(rownames(beta) %in% rownames(Islands.UCSC_beta_X==F)))
# 
# #check for any weird entries (perhaps X is denoted x)
# head(substr(Islands.UCSC$Islands_Name, 1,4))
# unique(substr(Islands.UCSC_beta$Islands_Name, 1,5))
# 
# betaminusXY <- beta[rownames(beta) %in% CpGxy==F,]
# #now beta has 412,612 sites--for global methyl

# #just x chromosome sites
# betaX <- beta[rownames(beta) %in% rownames(Islands.UCSC_beta_X)==T,]
# 
# #just y chromosome sites
# betaY <- beta[rownames(beta) %in% rownames(Islands.UCSC_beta_Y)==T,]

#correct way to identify x or y sites in beta matrix
library(ewastools)
library(data.table)
mani <- ewastools:::manifest_450K
data(IlluminaHumanMethylation450kmanifest)

# get a list of cpg sites that are on sex chromosomes
sexprobes <- mani %>%
  filter(chr %in% c("X","Y"))%>%
  select(probe_id)

Xprobes <- mani %>%
  filter(chr %in% "X")%>%
  select(probe_id)

Yprobes <- mani %>%
  filter(chr %in% "Y")%>%
  select(probe_id)

#get beta matrix (before remove gap probes) for only auto probes
beta_noXY <- beta[!rownames(beta) %in% sexprobes$probe_id,] #414,378 autosomal probes left
#beta matrix in terms of only x chrom sites
beta_Xonly <- beta[rownames(beta) %in% Xprobes$probe_id,] #9285 X sites in full beta
#beta matrix in terms of only Y chrom sites
beta_Yonly <- beta[rownames(beta) %in% Yprobes$probe_id,] #5 Y sites in full beta


#how can we check these are def autosomal?
mani %>% filter(probe_id %in% rownames(beta_noXY))%>%select(chr)%>%filter(chr %in% c("X","Y")) #this returns 0 cases--there are no sex probes in beta_noXY

  #Calc overall global DNAm (across single samples for each samp)

m <- colMeans(beta_noXY)
mpct <- m*100
head(colnames(m))

mpct_X <- colMeans(beta_Xonly)*100
mpct_Y <- colMeans(beta_Yonly)*100

colMeans(beta[,1:2])

x<-c("a","c")
y<-c("c","a")
z<-c("a","c")
all.equal(x,y)
all.equal(x,z)

#every probe site in beta matrix has entry in Islands.UCSC--know region of every site--NOT chrom location

genomic.region <- function(X, region, anno='450k'){
  #x=matrix with cpg as row names
  #region=area of interest
  #anno= 450k or EPIC
  # if (anno == '450k' | anno=='450K'){
  #   library(IlluminaHumanMethylation450kanno.ilmn12.hg19)} else if (anno=='epic'| anno=='EPIC'){
  #     library(IlluminaHumanMethylationEPICanno.ilm10b2.hg19)
  #     }else {
  #       stop('450k or EPIC only.')
  #     }

#data("Islands.UCSC")
  Islands.UCSC <- Islands.UCSC[rownames(X),]
  Islands.UCSC <- Islands.UCSC[Islands.UCSC$Relation_to_Island==region,]
return(X[rownames(Islands.UCSC),])
}
#unique(Islands.UCSC$Relation_to_Island)
# 
# # compute means by region
meanDNAm.sea <- colMeans(genomic.region(beta_noXY, 'OpenSea'))*100

#test that ordering is in same order as pdqc order so merge works
meanDNAm.seaTEST <- colMeans(genomic.region(beta_noXY[,match(pdqc$MethID,colnames(beta))], 'OpenSea'))*100

# check that same results 
all.equal(meanDNAm.sea, meanDNAm.seaTEST)

meanDNAm.shoren <- colMeans(genomic.region(beta_noXY,'N_Shore')) * 100
meanDNAm.shores <- colMeans(genomic.region(beta_noXY,'S_Shore')) * 100
meanDNAm.shelfn <- colMeans(genomic.region(beta_noXY,'N_Shelf')) * 100
meanDNAm.shelfs <- colMeans(genomic.region(beta_noXY,'S_Shelf')) * 100
#meanDNAm.shore <- colMeans(genomic.region(beta,'Shore')) * 100
#meanDNAm.shelf <- colMeans(genomic.region(beta,'Shelf')) * 100
meanDNAm.island <- colMeans(genomic.region(beta_noXY,'Island')) * 100


# try on own
# Calc global DNam for shore regions

#Islands.UCSC1 <- Islands.UCSC[Islands.UCSC$Relation_to_Island == #"N_Shore"|Islands.UCSC$Relation_to_Island == "S_Shore",]


#use dplyr on large dataframe with multiple classes
Islands.UCSC_shores <- as.data.frame(Islands.UCSC) %>% dplyr::filter(Relation_to_Island == "N_Shore"| Relation_to_Island == "S_Shore")
#class(Islands.UCSC[2])

m_shores <- colMeans(beta_noXY[rownames(beta_noXY) %in% rownames(Islands.UCSC_shores),])*100


# Global DNAm shelves calc
Islands.UCSC_shelves <- as.data.frame(Islands.UCSC) %>% dplyr::filter(Relation_to_Island == "N_Shelf"| Relation_to_Island == "S_Shelf")

m_shelves <- colMeans(beta_noXY[rownames(beta_noXY) %in% rownames(Islands.UCSC_shelves),])*100

# Check that order of calculation stays in same order as cols of beta matrix
m_shelvesTEST <- colMeans(beta_noXY[rownames(beta_noXY) %in% rownames(Islands.UCSC_shelves),match(pdqc$MethID,colnames(beta_noXY))])*100

#test if the m_shelves calculation is in order of pdqc MethID's so we can merge properly
all.equal(m_shelves,m_shelvesTEST)
# check that ordering of the 2 vectors is exact same
identical(m_shelves,m_shelvesTEST, attrib.as.set=F) #attrib.as.set=F looks that vectors are ordered
tail(m_shelves)
tail(m_shelvesTEST)

# Global DNAm north shelf check vs function
Islands.UCSC_Nshelf <- as.data.frame(Islands.UCSC) %>% dplyr::filter(Relation_to_Island == "N_Shelf")

m_shelves_Nshelf <- colMeans(beta_noXY[rownames(beta_noXY) %in% rownames(Islands.UCSC_Nshelf),])*100
#gives same results
# Check genomic region function results with my own
all.equal(meanDNAm.shelfn,m_shelves_Nshelf)

# DEBUGGING practice
#dim(beta[rownames(Islands.UCSC_Nshelf),])
# Before, had m_shelves_NT <- colMeans(beta[rownames(Islands.UCSC_Nshelf)) %in% rownames(beta),]) -- wasn't giving right calculations


# figure out number of probes in each region (open sea, shore, shelf)
library(gt)


as.data.frame(table(Islands.UCSC_beta$Relation_to_Island)) %>% gt() %>% cols_label(Var1="Region",Freq="Probe Number")

#proportions
prop.table(table(Islands.UCSC_beta$Relation_to_Island))
as.data.frame(round(prop.table(table(Islands.UCSC_beta$Relation_to_Island)),2)) %>% gt() %>% cols_label(Var1="Region",Freq="Proportion")



#do the same but compare between gap and non gap probes
# as.data.frame(table(Islands.UCSC_beta[gapprobes820,]$Relation_to_Island)) %>% gt() %>% cols_label(Var1="Region",Freq="Probe Number")
# 
# as.data.frame(round(prop.table(table(Islands.UCSC_beta[gapprobes820,]$Relation_to_Island)),2)) %>% gt() %>% cols_label(Var1="Region",Freq="Proportion")

#non-gap probes
# as.data.frame(table(Islands.UCSC_beta[which(rownames(Islands.UCSC_beta) %in% gapprobes820 == F),]$Relation_to_Island)) %>% gt() %>% cols_label(Var1="Region",Freq="Probe Number")
# 
# as.data.frame(round(prop.table(table(Islands.UCSC_beta[which(rownames(Islands.UCSC_beta) %in% gapprobes820 == F),]$Relation_to_Island)),2)) %>% gt() %>% cols_label(Var1="Region",Freq="Proportion")

#chisq test for difference
# (tabbo=table(Islands.UCSC_beta[which(rownames(Islands.UCSC_beta) %in% gapprobes820 == F),]$Relation_to_Island,Islands.UCSC_beta[gapprobes820,]$Relation_to_Island))
# (chisq.test(tabt2)) 


# put global methylation values into dataframe
#t <- as.data.frame(m)
#head(rownames(t))
#sum(duplicated(m))
#t1 <- t %>% mutate(MethID=rownames(t))

#combine globalDNAm measures into dataframe
df <- data.frame(m,mpct,mpct_X,mpct_Y, m_shelves,m_shores,meanDNAm.island, meanDNAm.sea, meanDNAm.shoren, meanDNAm.shores, meanDNAm.shelfn, meanDNAm.shelfs)
head(rownames(df))

#label each globalDNAm calc with correct id names
df1 <- df %>% mutate(MethID=rownames(df))

# merge pheno & global methylation values (was pdqcA)
pdqc2a <- left_join(pdqcA1,df1,by ="MethID") # only child age 9 pheno & global meth values (including duplicate kids)

# if want: could do groupby(methid) and get mean methylation value for the duplicate samples
```


```{r}
# select only variables of interest
mainvars_age9_pd_1 <- pdqc2a %>%
  select(1:10,16:33)

# add cell props
# calculate cell proportions
library(EpiDISH)
#beta with gaps and sex chrom probes still in 
beta <- beta[,match(mainvars_age9_pd_1$MethID,colnames(beta))]
cells <- epidish(beta.m=beta, ref.m=centEpiFibIC.m)

# merge these new cell proportions with mainvars_9na
mv_9_try <- data.frame(mainvars_age9_pd_1,cells$estF%>% data.frame)

# pdqc vars that we want to merge by
pd <-c("m1city","cm1bsex","cm1ethrace","cm1inpov", "cm5b_age","cm5b_ageyrs","cm5povco","ck6ethrace")

fullphen %>%
  select(contains("intmon"))

# full phen vars of interest
mainvars_age9_fp_1 <- fullphen %>%
  select("id","cf1ethrace","cm1edu","cf1edu","hv5_cbmi","hv5_mbmi","IH5mombmi","hv5_ppvtss","hv5_ppvtraw", "m1g1", "m1g4", "k5conf2","m1intmon","m1intyr","cm5intmon","cm5intyr","hv5_intmon","hv5_intyr","o5_intmon","o5_intyr","p5_intmon","p5_intyr","hv5_agem", pd)
# add cm1inpov see what happens--merge data has cminpov.x and cominpov.y


# need to keep all variables
mainvars9_a <- merge(mainvars_age9_fp_1,mv_9_try, by.x=c("id",pd),by.y=c("idnum",pd), all=T) 
  #merge(mainvars_age9_fp_1,mainvars_age9_pd_1, by.x = #"id",by.y="idnum", all.x=TRUE) ** had this before but PROBLEM was that it would appear that a case didnt  have a variable since it had no methID even tho it truly did in the fullpheno data
# before: delete pd in the by.x and by.y

# checking for duplicates (all say 25)
sum(duplicated(mv_9_try$id)) # 25 duplicate ids
(sum(duplicated(mainvars9_a$id)))
sum(mainvars9_a$crossdupe, na.rm=TRUE)


# turn all negative #s to NAs (except the k5conf2 variable because we only eliminate cases with -9 value -- not -6 or -3)
mainvars9_a <- mainvars9_a %>% mutate_at(vars(-k5conf2),funs(replace(.,.<0,NA)))


# this might have actually worked
mainvars9_a1 <- mainvars9_a %>%
  mutate(whymissing=ifelse(k5conf2==-9,"no wave5",
                           ifelse(is.na(MethID), "no methyl",
                                  ifelse(crossdupety==TRUE, "duplicate",
                                         ifelse(is.na(cm1bsex),"no sex",ifelse(is.na(hv5_agem),"no age",
                                          #ifelse(is.na(hv5_ppvtss),"no PPVT",
                                                       ifelse(is.na(cm1ethrace), "no mom race",
                                                        ifelse(is.na(cm1edu),"no mom edu",
                                                              ifelse(is.na(hv5_cbmi),"no child BMI", ifelse(is.na(cm1inpov),"no povratio", ifelse(is.na(m1g1),"no mom health", ifelse(is.na(m1g4), "no mom smoke", NA))))))))))))

table(mainvars9_a1$whymissing)

FIN <- mainvars9_a1 %>%
  filter(is.na(whymissing)) #n=820 as expected

#Factor key categorical variables
 FIN$m1city <- factor(FIN$m1city)
 FIN$cm1bsex <- factor(FIN$cm1bsex)
 FIN$cm1edu <- factor(FIN$cm1edu)
 FIN$cf1edu <- factor(FIN$cf1edu)
 FIN$cm1ethrace <- factor(FIN$cm1ethrace)
 FIN$cf1ethrace <- factor(FIN$cf1ethrace)
 FIN$m1g1 <- factor(FIN$m1g1)
 FIN$m1g4 <- factor(FIN$m1g4)

# Trial and Error: inclusion exclusion table
table(is.na(mainvars9_a$cm1edu)) #what do you expect?
#mainvars9_a$whymissing_kelly<-ifelse()
table(mainvars9_a$whymissing_kelly) #what do you get? Do they match?

# fill table with random filler "z" for now
mainvars9_a$whymissing_kelly <- 'z'

table(mainvars9_a$k5conf2!=-9) # 1498 are not in wave
mainvars9_a$whymissing_kelly<-ifelse(mainvars9_a$k5conf2==-9,"no wave5",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly) #1498---leaves us with 3400

# of this 3400--how many are missing methylation? 
#figure out who has "no methyl" and "no wave5"
#mainvars9_a

table(is.na(mainvars9_a$MethID)) # 866 people have methID
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$MethID), "no methyl",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

# Note: 1 person has methyl data but not in wave 5!!
head(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$crossdupe))
mainvars9_a$whymissing_kelly<-ifelse(mainvars9_a$crossdupety==TRUE, "duplicate",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

#why did all the nomethyl's get erased ..?--no wave5 still there
pickone <- mainvars9_a %>%
  filter(crossdupe==T)%>%
  group_by(duplicated(paste0(id,childteen)))
sum(duplicated(paste0(mainvars9_a$id,mainvars9_a$childteen)))


table(is.na(mainvars9_a$cm1bsex))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$cm1bsex),"no sex",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$hv5_ppvtss))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$hv5_ppvtss),"no PPVT",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$cm1ethrace))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$cm1ethrace), "no mom race",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$cm1edu))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$cm1edu),"no mom edu",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$hv5_cbmi))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$hv5_cbmi),"no child BMI",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$cm1inpov))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$cm1inpov),"no pov ",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$m1g1))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$m1g1),"no mom health ",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

table(is.na(mainvars9_a$m1g4))
mainvars9_a$whymissing_kelly<-ifelse(is.na(mainvars9_a$m1g4),"no mom smoke",mainvars9_a$whymissing_kelly) 
table(mainvars9_a$whymissing_kelly)

sum(mainvars9_a$whymissing_kelly=="z",na.rm=T) #n=820!

# try to find which cases are missing both 
mainvars9_a %>%
  filter(whymissing_kelly == "no methyl" |whymissing_kelly == "no wave5")%>%
  filter(whymissing_kelly == "duplicate")

# get n=820 final dataset
mvFIN <- mainvars9_a %>%
  filter(mainvars9_a$whymissing_kelly=="z")

#factor categorical vars
#Factor key categorical variables
 mvFIN$m1city <- factor(mvFIN$m1city)
 mvFIN$cm1bsex <- factor(mvFIN$cm1bsex)
 mvFIN$cm1edu <- factor(mvFIN$cm1edu)
 mvFIN$cf1edu <- factor(mvFIN$cf1edu)
 mvFIN$cm1ethrace <- factor(mvFIN$cm1ethrace)
 mvFIN$cf1ethrace <- factor(mvFIN$cf1ethrace)
 mvFIN$m1g1 <- factor(mvFIN$m1g1)
 mvFIN$m1g4 <- factor(mvFIN$m1g4)
 

mainvars9_a$whymissing[is.na(mainvars9_a$cm1edu)==T]
  
# Table showing number of cases and reason missing
table(mainvars9_a$whymissing_kelly)

#sum the cases that have no methylation or have duplicates
sum(!is.na(mainvars9_a$MethID)) # 866
sum(mainvars9_a$crossdupe==T, na.rm=T) # 25 dups
sum(mainvars9_a$whymissing_kelly == "duplicate")
sum(mainvars9_a$whymissing == "no child BMI")
sum(is.na(mainvars9_a$hv5_cbmi))

# mainvars9_a %>%
#   filter(is.na(whymissing_kelly))%>%
#   count() #820


table(mainvars9_a$whymissing_kelly)
#summary(finalmainvars)

#check that mvFIN and FIN have same exact children in them
#all.equal(mvFIN$id, FIN$id) TRUE


# fill in table--we using k5conf2 variable to eleminate kids that don't have wave 5 (age 9) data
fullphen %>%
  filter(k5conf2==-9)%>%
  select(hv5_ppvtss)

# just get down to 2580 for table
 # get idnums of the 820 and get rid of correspnoding idnumbs in the 3400 dataset
# may need to remove
MVTRY <- fullphen %>%
 filter(k5conf2!=-9)

# select idnums of those in our final analysis dataset
idnums = c(FIN$id)
head(idnums)

# excluded sample data
ah <- MVTRY %>%
  select("id","cm1bsex", "cf1ethrace","cm1edu","cf1edu","hv5_cbmi","hv5_mbmi","hv5_ppvtss","m1g1", "m1g4", "k5conf2","cm1bsex","cm1ethrace", "m1city","hv5_agem","cm1inpov")%>%
  filter(!(id %in% idnums))
#%>%filter(id %in% idnumsdup) #used to check that none of duplicates data were counted in excluded samples


ah <- ah %>% mutate_at(vars(-k5conf2),funs(replace(.,.<0,NA)))


#Factor key categorical variables
 ah$m1city <- factor(ah$m1city)
 ah$cm1bsex <- factor(ah$cm1bsex)
 ah$cm1edu <- factor(ah$cm1edu)
 ah$cm1ethrace <- factor(ah$cm1ethrace)
 ah$m1g1 <- factor(ah$m1g1)
 ah$m1g4 <- factor(ah$m1g4)

summary(ah)
#keep adding pheno variables
table(ah$cm1ethrace)
table(ah$cm1bsex)
table(ah$cm1)
table(ah$m1city)
table(ah$cm1edu)
sd(ah$cm1inpov)
sd(ah$hv5_mbmi)
sum(is.na(ah$hv5_mbmi))
sum(is.na(ah$hv5_ppvtss))
sum(is.na(ah$cm1bsex))
sum(is.na(ah$cm5b_age))
sum(is.na(ah$cm1inpov))
t2 <- mainvars9_a %>%
  filter(!is.na(mainvars9_a$whymissing_kelly))
  #filter(crossdupe==F)

```

```{r}
# exclusion inclusion diagram
# data removing kids not in wave
MVTRY0 <- fullphen %>%
 filter(k5conf2!=-9)%>%
  select("id","cf1ethrace","cm1edu","cf1edu","hv5_cbmi","hv5_mbmi","IH5mombmi","hv5_ppvtss","m1g1", "m1g4", "k5conf2", "hv5_agem")
#3400 9 year olds


#id's for pdqcA (just children age 9 with methyl data n=866) 
idnums3 <- c(pdqcA$idnum)

#id's for pdqcAA (just children age 9 with nonduplicate methyl data -- n=841)
idnumstry <- c(pdqcAA$idnum)

#data for only children with methylation data
MVTRY2 <- MVTRY %>%
filter(id %in% idnums3) #why is this n=840 also

# data for only children with NONDUP methylation data
MVTRY3 <- MVTRY %>%
filter(id %in% idnumstry)
# are any duplicate id now

# there are no duplicate ids but could be duplicate id & child)? no there couldn't be? 
sum(duplicated(MVTRY2$id))
sum(duplicated(paste0(pdqcA$childteen,pdqcA$idnum)))
sum(is.na(MVTRY2$hv5_mbmi))

# merge : think through how to 
#MVTRY3 <- merge(MVTRY2,mv_9_try, by.x="id",by.y="idnum", all=T) 
MVTRY3 <- merge(MVTRY2,mv_9_try, by.x = "id", by.y="idnum", all.x=TRUE)
#MVTRY3 %>%
  #filter(crossdupe==F)

#checking that both methods of attaining n=820 worked
#match(idnums, idnums2)
```

Repeat with working FIN data (n=820)
```{r}
# just get down to 2580 for table
 # get idnums of the 820 and get rid of correspnoding idnumbs in the 3400 dataset
# may need to remove
# MVTRY <- fullphen %>%
#  filter(k5conf2!=-9)
# 
# #idnums2 = c(FIN$id)
# #head(idnums2)
# 
# #identical(idnums, idnums2)
# 
# ah2 <- MVTRY %>%
#   select("id","cf1ethrace","cm1edu","cf1edu","hv5_cbmi","hv5_mbmi","IH5mombmi","hv5_ppvtss","m1g1", "m1g4", "k5conf2","cm1bsex","cm1ethrace", "m1city","cm5b_age","cm1inpov","hv5_agem")%>%
#   filter(!(id %in% idnums))
# 
# ah2 <- ah2 %>% mutate_at(vars(-k5conf2),funs(replace(.,.<0,NA)))
# 
# #Factor key categorical variables
#  ah2$m1city <- factor(ah2$m1city)
#  ah2$cm1bsex <- factor(ah2$cm1bsex)
#  ah2$cm1edu <- factor(ah2$cm1edu)
#  ah2$cm1ethrace <- factor(ah2$cm1ethrace)
#  ah2$m1g1 <- factor(ah2$m1g1)
#  ah2$m1g4 <- factor(ah2$m1g4)

#summary(ah2)
#sd(ah2$cm1inpov)
#sd(ah2$hv5_agem, na.rm=T)
#mean(ah2$hv5_agem,na.rm=T)
```

More efficient way for inclusion/exclusion table
```{r}
cat_vars = c("cm1ethrace","cm1edu","m1g1","m1g4")
cat_inc = subset(FIN, select=cat_vars)
cat_exc = subset(ah, select=cat_vars)

# get proportions of each category for all categ. variables
cat_prop_inc = t(apply(cat_inc, 2, function(x) prop.table(table(x))))
cat_prop_exc = t(apply(cat_exc, 2, function(x) prop.table(table(x))))

#new dataset with inc variable (whether the row was included or excluded from final sample)
ah3 <- MVTRY %>%
  select("id","cf1ethrace","cm1edu","cf1edu","hv5_cbmi","hv5_mbmi","IH5mombmi","hv5_ppvtss","m1g1", "m1g4", "k5conf2","cm1bsex","cm1ethrace", "m1city","cm5b_age","hv5_agem","cm1inpov")%>%
  mutate(inc = ifelse(!(id %in% idnums), "exc","inc"))

#change the negative values to NA for ease
ah3 <- ah3 %>% mutate_at(vars(-k5conf2),funs(replace(.,.<0,NA)))

ah3$m1city <- factor(ah3$m1city)
ah3$cm1ethrace <- factor(ah3$cm1ethrace)
ah3$cm1edu <- factor(ah3$cm1edu)
ah3$m1g1 <- factor(ah3$m1g1)


#try
pval = sapply(1:length(cat_vars),function(x) chisq.test(table(ah3[,x],ah3$inc),correct=F)$p.value)

# Hand check cm1ethrace chi-sq test value from line above
(tabt=table(ah3$cm1ethrace,ah3$inc))
(chisq.test(tabt)) # should be accurate

(tabt2=table(ah3$cm1edu,ah3$inc))
(chisq.test(tabt2)) # should be accurate

# says "chi-squared approx may be incorrect)
(tabt3=table(ah3$m1g1,ah3$inc))
(chisq.test(tabt3))
(fisher.test(tabt3))#instead do a fishers exact test


(tabt4=table(ah3$m1g4,ah3$inc))
(chisq.test(tabt4))
(fisher.test(tabt4))

(tabt5=table(ah3$m1city,ah3$inc))
(fisher.test(tabt5,simulate.p.value=T))# weird error
# Attempt to hand-do cm1ethrace chi-sq test
#t <- matrix(c(160,452,179,29, 541,1261,687,84),ncol=4, byrow=T)
#chisq.test(t)
```



Get idnums of final dataset for Jonah--verify no related individs
```{r Creating idnums.txt for relatedness}
wut <- data.frame(idnums, idnums)
write.table(wut, file="/home/alreiner/Projects/ffcw/idnums2.txt", sep= ' ',quote=F, col.names=F, row.names=F)
save(idnums,file="/home/alreiner/Projects/ffcw/idnums.txt")
```


IncExc Diff in Props
```{r}
#Categorical Vars
prop.test(x=c(417,1364),n=c(820,2580)) #cm1bsex
prop.test(x=c(160,541),n=c(820,2580)) #cm1ethrace


# Continuous Vars
t.test(ah$hv5_ppvtss,FIN$hv5_ppvtss, alternative="two.sided", var.equal = TRUE) # cog
var.test(ah$hv5_ppvtss,FIN$hv5_ppvtss, alternative="two.sided", var.equal = TRUE) # cog

t.test(ah$cm1inpov,FIN$cm1inpov, alternative="two.sided", var.equal = TRUE) # pov_baseline
var.test(ah$cm1inpov,FIN$cm1inpov, alternative="two.sided", var.equal = TRUE) # pov_baseline

t.test(ah$hv5_cbmi,FIN$hv5_cbmi, alternative="two.sided", var.equal = TRUE) # child bmi
var.test(ah$hv5_cbmi,FIN$hv5_cbmi, alternative="two.sided", var.equal = TRUE) # child bmi

t.test(ah$hv5_mbmi,FIN$hv5_mbmi, alternative="two.sided", var.equal = TRUE) # mom bmi
var.test(ah$hv5_mbmi,FIN$hv5_mbmi, alternative="two.sided", var.equal = TRUE) # mom bmi

# Continuous Vars
t.test(ah$hv5_agem,FIN$hv5_agem, alternative="two.sided", var.equal = TRUE) # age
var.test(ah$hv5_agem,FIN$hv5_agem, alternative="two.sided", var.equal = TRUE) # age

#Check underlying distribution of the variables
library("psych")
nums1 <- unlist(lapply(FIN,is.numeric)) #nums is only numeric variables 

# histograms and correl for included samples
pairs.panels(FIN[,c("hv5_ppvtss","cm1inpov","hv5_agem","hv5_cbmi", "hv5_mbmi","m","Epi","Fib","IC")])

hist(FIN$m)

# histograms and correl for excluded samples
pairs.panels(ah[,c("hv5_ppvtss","cm1inpov","hv5_agem","hv5_cbmi", "hv5_mbmi")])

# histograms and correl for excluded samples
pairs.panels(ah[,c("hv5_ppvtss","cm1inpov","hv5_agem","hv5_cbmi", "hv5_mbmi")])

```

Final Analysis Dataset INC-EXC Table
```{r}
sum(is.na(FIN$hv5_ppvtraw))
summary(FIN)
mean(FIN$hv5_agem, na.rm=T)
sd(FIN$hv5_agem, na.rm=T)
```
Find the city identifiers
```{r}
table(fullphen$m1city)
```


Hand Calculate age9 for those missing
```{r Unnecessary Age Imputation}
# intitial date and year: m1intmon & m1intyr)
# fullphen %>%
#   select(contains("intmon"))
# class(fullphen$cm5intmon) # mom's interview month
# class(fullphen$cm5intyr) #mom's interview year
# 
# library(zoo)
# library(lubridate)
# 
# # Check which variable is closest to cm5intyr
# sum(FIN$cm5intyr == FIN$hv5_intyr, na.rm=T) #747 cases where the same year
# sum(FIN$cm5intyr == FIN$o5_intyr, na.rm=T) #748
# sum(FIN$cm5intyr == FIN$p5_intyr, na.rm=T) #773 cases that are the same year
# sum(FIN$cm5intmon == FIN$p5_intmon, na.rm=T) #535 cases that are the same year
# sum(FIN$cm5intmon == FIN$o5_intmon, na.rm=T) #517 cases that are the same year
# # SO let's use the p5_intyr variable since the most cases share the same year as cm5_intyr
# 
# FIN$IntDateM9 <- as.yearmon(paste(FIN$p5_intyr, FIN$p5_intmon),"%Y %m") #p5 is when PCG was interviewed
# FIN$IntDateM1 <- as.yearmon(paste(FIN$m1intyr, FIN$m1intmon),"%Y %m")
# #FIN$cm5b_age_byhand <- interval(FIN$IntDateM1, FIN$IntDateM9) %/% #months(1)
# FIN$cm5b_age_byhand <- time_length(interval(FIN$IntDateM1, FIN$IntDateM9), unit="month")
# ##class(FIN$IntDateM1)
# class(FIN$cm5b_age_byhand )
# 
# # Check the direction of difference (cm5b_age - imputedage)
# FIN$dirdif = FIN$cm5b_age - FIN$cm5b_age_byhand
# summary(FIN$dirdif) # min=0, max=3 (all positive--so imputed age always underestimates true age (cm5bage))
# plot(FIN$dirdif, ylab="Difference between true age and imputed age in months")
# # find percentage of kids for whome imputed age does not euqal true age (not including n=21 missing cm5b_age obvi)
# sum(FIN$dirdif != 0, na.rm=T)
# sum(FIN$dirdif == 0, na.rm=T)
# table(factor(FIN$dirdif))
# 
# #get the id's of folks with missing cm5b_age so we can plot their estimated age
# which(is.na(FIN$cm5b_age))
# 
# FIN1 <- FIN %>% dplyr::filter(is.na(FIN$cm5b_age)==T)
# 
# #table(FIN$cm5intyr, FIN$cm5b_age)
# #
# 
# 
# plot(FIN1$cm5b_age_byhand)
# FIN[is.na(FIN$cm5b_age),]
# plot(FIN[is.na(FIN$cm5b_age),"cm5b_age_byhand"])
# range(FIN[is.na(FIN$cm5b_age),"cm5b_age_byhand"])
# 
# 
# #plot a figure the by hand age calculation, with red points denoting those n=21 kids missing cm5b_age
# plot(FIN$cm5b_age_byhand, col=ifelse(is.na(FIN$cm5b_age), "red","black"), ylab="Age in Months by hand", title = "Imputed Age for n=820 Age 9 Children")
# #, xlab="Age BY HAND",ylab="True Age")
# 
# 
# # plot of the true age for the kids without missing age (n=799)and the imputed age for kids with missing cm5b_age (n=21)
# plot(ifelse(!is.na(FIN$cm5b_age),FIN$cm5b_age, FIN$cm5b_age_byhand), col=ifelse(is.na(FIN$cm5b_age), "red","black"), ylab="True Age in Months relative to imputed age for n=21 cases")
# 
# 
# #all(FIN$cm5b_age == FIN$cm5b_age_byhand, na.rm=T)
# #which(!all(FIN$cm5b_age != FIN$cm5b_age_byhand, na.rm=T))
# 
# FIN %>% select(cm5b_age, cm5b_age_byhand)
# 
# # check how accurate each age_by_hand calculation is! --max is 3 months off
# max(abs(FIN$cm5b_age_byhand-FIN$cm5b_age), na.rm=T)
# 
# 
# plot(FIN$cm5b_age_byhand,FIN$cm5b_age, xlab="Age BY HAND",ylab="True Age")+ geom_abline(intercept=0, slope=1)
# 
# ageplot <- ggplot(FIN, aes(x=cm5b_age_byhand,y=cm5b_age))+geom_point()+ xlab("Imputed Age in months")+ ylab("True Age in months")+ geom_abline(intercept=0, slope=1)
# ageplot
```
Need to use a different age variable
```{r}
# store the id's of kids with missing cm5b_age values (n=21)
# idsFINmiss <- c(FIN1$id)
# 
# #check what the original values were for these 21 kids (was -9--meaning not in wave)
# mainvars_age9_fp_1 %>%
#   filter(id %in% idsFINmiss) 
# 
# #realized we should be using hv5_agem variable and not cm5b_age
# lulu <- fullphen %>%
#   select(id, hv5_agem)%>%
#   filter(id %in% idnums)
#   #ch5agem
# sum(is.na(lulu$hv5_agem)) #none of our n=820 kids are missing this! yay!

```

Stacked barplot of cell type proportions across samples
```{r}
hmer <- mvFIN %>% gather(key=Type, value=Percent, 48:50)%>% arrange(desc(Percent))

hmer$id <- factor(hmer$id, levels = unique(hmer$id))
ggplot(data=hmer, aes(x=id, y=Percent, fill=Type))+geom_bar(stat="identity")+theme(axis.text.x = element_blank(),axis.ticks = element_blank())+ggtitle("Cell Type Composition for n=820 samples")
```




write csv for table Rmd
```{r}
mvFIN_clean <-
  mvFIN %>% select(-c(whymissing_kelly, min, probe_fail_pct, crossdupe, 21:30, cm5b_age, cm5b_ageyrs),flag_sex)

mvFIN_clean <- mvFIN_clean %>%
  filter(flag_sex == FALSE)

#join mv_fin_clean with batch/slide variables


write.csv(mvFIN_clean, file="./data/analysis1data", row.names = FALSE)

# are any wrong predicted sex id's in here
sum(idnums_sex %in% mvFIN_clean$id) # 12/13 kids with wrong sex are in analysis data
head(idnums_sex)

#what are cell proportion estimates when exclude gap probes and sex chroms

```


```{r}
#gaphunter

#820
gaps820 <- gaphunter(beta[,match(mvFIN_clean$MethID,colnames(beta))], threshold=0.05, keepOutliers = FALSE, outCutoff=0.01, verbose=TRUE)

gapprobes820 <- c(rownames(gaps820$proberesults))
head(gaps820$proberesults)
```


15 year olds
```{r}
# get rid of the moms and children in pdqc
pdqcTeen <- pdqc %>%
  dplyr::filter(childteen != "C") %>%
  filter(childteen != "M")
#n=866 (1 case is not in wave)

# returns logical T/F if any duplicate C and idnum combinations (paste0--combines C/T and idnum)
pdqcTeenA <- pdqcTeen %>%
  mutate(crossdupe1 =duplicated(paste0(pdqcTeen$childteen,pdqcTeen$idnum)),
         flag_sex1 = predicted_sex != sex
         )%>%
  select(1:19, -cm5intyr, -cm5intmon, -cmf5finjail,crossdupe1, probe_fail_pct,flag_sex1)

library(readr)
analysis1data <- read_csv("/home/alreiner/Projects/ffcw/data/analysis1data")

nineyoID<-as.character(c(analysis1data$id))
pdqcTeenA_match <- pdqcTeenA %>% filter(idnum %in% nineyoID) #9 year olds that have 15 year old data (note: could be duplicate entries in this data)
nineyo_match <- c(pdqcTeenA_match$idnum)

#816 of the idnums in our idnums are in la2...so 8 duplicates
#find which 8 are duplicates and keep just 1

sum(nineyoID %in% nineyo_match) #787 of the n=808 9 year olds have 15 year old data
identical(nineyoID,nineyo_match) #not identical

#for duplicate samples, find which had lower probe_fail_pct and keep that one
pdqcTeenA1 <- pdqcTeenA_match %>%
group_by(idnum)%>%
  mutate(min = min(probe_fail_pct))%>%
  arrange(probe_fail_pct)%>%
  mutate(crossdupety = duplicated(paste0(childteen,idnum))) %>%select(idnum, probe_fail_pct, crossdupety,childteen,min,crossdupe1)

dups15 <- pdqcTeenA1 %>%
  filter(crossdupe1==T) #29 duplicates
dups15 %in% nineyoID

# save idnums of these technical duplicates (n=29)
idnumsdup15 <- c(dups15$idnum)

#check how many duplicate C,idnum (n=29)
sum(duplicated(pdqcTeenA1$idnum))

#view probe fail pct for these idnums
g <- pdqcTeenA1 %>%
  filter(idnum %in% idnumsdup15)%>%
  select(idnum, probe_fail_pct, crossdupety,childteen)

#keep entries where crossupety=F (smaller probe fail pct)
pdqcTeen_nodup <- pdqcTeenA1%>%filter(crossdupety==F)
#final check for no duplicates
sum(duplicated(pdqcTeen_nodup$idnum))

write.csv(pdqcTeen_nodup, file="/home/alreiner/Projects/ffcw/data/pdqcTeen_nodup_787.csv", row.names = FALSE)

####gap hunter 15 year olds
gaps787 <- gaphunter(beta[,match(mvFIN_clean$MethID,colnames(beta))], threshold=0.05, keepOutliers = FALSE, outCutoff=0.01, verbose=TRUE)

gapprobes820 <- c(rownames(gaps820$proberesults))

```

double check 15 year olds
```{r}
#filter down to analysis id--n=816 kids
la2<-as.character(c(analysis1data$id))
pdqcTRY <- pdqcTeen %>%
  filter(idnum %in% la2)

#see how many are duplicates (n=29 duplicates in this 816 sample)
pdqcTRY1 <- pdqcTRY %>%
  mutate(crossdupe1 =duplicated(paste0(pdqcTRY$childteen,pdqcTRY$idnum)))
  
sum(pdqcTRY1$crossdupe1==T)
```